codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java

```diff
@@ -715,22 +731,44 @@ MethodSpec buildEventObservableFunction(
                         .build())
                 .build();
 
+        observableMethodBuilder
+                .addStatement("return web3j.ethLogObservable(filter).map($L)", converter);
```

Comment:
The generated code needs to be Java 1.6 compatible - the reason being that smart contract wrappers for Java and Android are generated using exactly the same process.

Reply:
I don't see where the generated code is not Java 6 compatible 😕 

The generated code for this specific line is perfectly fine:
```
return web3j.ethLogObservable(filter).map(new Func1<Log, TransferEventResponse>() {
   ...
});
```
Am I missing something?

Reply:
There's however an incompatibility in the `_addresses` field initialiser:
```
static {
    _addresses = new HashMap<>();
    _addresses.put("4", "0xaea9d31a4aeda9e510f7d85559261c16ea0b6b8b");
    _addresses.put("4447", "0x254dffcd3277c0b1660f6d42efbb754edababc2b");
}
```
This should be `_addresses = new HashMap<String, String>();`

Reply:
Apologies my bad!